src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -114,9 +133,15 @@ export const NavBarSchedule: FC<NavBarScheduleProps> = ({ isVisible }) => {
   );
 
   const schedule: ScheduleDay = useMemo(() => {
-    const dayStart = addDays(startOfToday(), selectedDayIndex);
+    const dayStart = addDays(firstDay, selectedDayIndex);
     const daysEvents = relatedVenueEvents
-      .filter(isEventWithinDate(selectedDayIndex === 0 ? Date.now() : dayStart))
+      .filter(
+        isEventWithinDate(
+          selectedDayIndex === 0 && !hasScheduleStartDate
+            ? Date.now()
+            : dayStart
+        )
+      )
```

Comment:
Pending answer to https://github.com/sparkletown/sparkle/pull/1419#discussion_r635719319 RE: the purpose of adding `!hasScheduleStartDate` here.

Reply:
Looking at the code above this.

```typescript
// Original Code
const dayStart = addDays(startOfToday(), selectedDayIndex);

// Current PR Code
const dayStart = addDays(firstDay, selectedDayIndex);
```

I believe the usage of `Date.now` in this filter is irrelevant.

In the original code we used `startOfToday()` + `Date.now()`, which are both going to represent the same day.

Since the first item on the schedule has index `0`, when it is selected, `dayStart` is going to be set to essentially:

```typescript
// Original Code
const dayStart = addDays(startOfToday(), 0);

// Current PR Code
const dayStart = addDays(firstDay, 0);
```

And since adding `0` days to the date is a noop, we essentially get:

```typescript
// Original Code
const dayStart = startOfToday();

// Current PR Code
const dayStart = firstDay;
```

Looking at `isEventWithinDate`, it already uses `startOfDay`/`endOfDay` on the provided `checkDate` value, meaning so long as `checkDate` represents the same date, it doesn't matter what time of day it is when passed in:

- https://github.com/sparkletown/sparkle/blob/staging/src/utils/event.ts#L65-L76

Based on this, it shouldn't matter if we're checking todays date, or a future date, and so should be able to just do the following:

```suggestion
      .filter(isEventWithinDate(startOfSelectedDay))
```

Reply:
Which it seems I already wrote in my review on the original feature.. I knew i'd written it somewhere already.. https://github.com/sparkletown/sparkle/pull/1302/files#r634036719

Reply:
Good catch — fixed

Reply:
Cool :)

Note that I think there might be slight tweaks to this again as part of @sunny-viktoryia's https://github.com/sparkletown/sparkle/pull/1426 PR that is currently in review (which is the cleanup/follow-on PR from #1302)

So depending on which gets merged first between it and this PR, we'll need to make sure we don't overwrite/lose any of the tweaks/changes.

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1419/commits/b78373170c83bbc19324c8694b04ea35b9ab06ef